src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -957,7 +958,9 @@ public bool WaitForTaskToBeDone(Action task)
                     try
                     {
                         _waitForTaskToBeDone.Reset();
-                        _task += task;
+                        Debug.Assert(_task == null);
+                        _exception = null;
+                        _task = task;
```

Comment:
Why did we switch from ` _task += task;` to `_task = task` ?

Reply:
In the beginning, I thought to allow adding more than one task but I realize this can lead to a problem in error handling when the lock owner runs the tasks and there is more than one and one of them throw an exception. And because we don't need it now I replace it to `=` to be more explicit about the usage and I added `Debug.Assert(_task == null);`